app/src/main/java/ch/sdp/vibester/activity/TypingGameActivity.kt

```diff
@@ -211,11 +132,84 @@ class TypingGameActivity : AppCompatActivity() {
         }
     }
 
+    override fun onDestroy() {
+        if(runnable !=null ){
+            h.removeCallbacks(runnable!!);}
+        if(this::gameManager.isInitialized){
+            gameManager.stopMediaPlayer()
+        }
+        super.onDestroy()
+    }
+
+    /**
+     * Print Toast message to announce the user if he wons or not
+     */
+    private fun hasWon(ctx: Context, score:Int, hasWon: Boolean, itwas: Song){
+        if(hasWon){
+            Toast.makeText(ctx,score.toString()+" Well Done!",Toast.LENGTH_SHORT).show()
+        }else{
+            Toast.makeText(ctx,"Sadly you're wrong, it was: "+itwas.getTrackName()+" by "+itwas.getArtistName(),Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    /**
+     * Generate a change of intent at the end of the game
+     */
+    fun checkAnswer(ctx: Context, choosenSong: Song?, gameManager: GameManager){
+        val playedSong = gameManager.getCurrentSong()
+
+        if(choosenSong != null && choosenSong.getTrackName() == playedSong.getTrackName() && choosenSong.getArtistName() == playedSong.getArtistName()){
+            gameManager.increaseScore()
+            gameManager.addCorrectSong()
+            hasWon(ctx, gameManager.getScore(),true,playedSong)
+        }else{
+            hasWon(ctx, gameManager.getScore(),false, playedSong)
+        }
+        playRound(ctx, gameManager)
+    }
+
+    /**
+     * Create the frame layout and its logic of the suggestion when user is typing
+     */
+    fun guess(song: Song, guessLayout: LinearLayout, ctx: Context, gameManager: GameManager): FrameLayout{
+        val frameLay = FrameLayout(ctx)
+        frameLay.background = borderGen(ctx)
+
+        // Horizontal Linear Layout to put Images and Text next one another
+        val linLay = LinearLayout(ctx)
+        linLay.setHorizontalGravity(1)
+        linLay.gravity = Gravity.LEFT
+
+        //Generate and add its component
+        linLay.addView(generateImage(song, ctx))
+        linLay.addView(generateSpace(100,100, ctx))
+        linLay.addView(generateText(song.getArtistName() + " - " + song.getTrackName(), ctx))
+
+        frameLay.addView(linLay)
```

Comment:
Is it possible to put the setting of linLay in a separate function? I feel like this function is doing a lot of things.